feat: add getProductImageFolder method to return product image base URL

ShaunakMishra25 opened the pull request:
### What
- Added a method `getProductImageFolder` to the `OpenFoodFacts` class.
- This method returns the correct image folder path based on the product barcode, following the OpenFoodFacts 3-digit split convention.

### Example

For barcode `6111035000430`, this method returns:
https://images.openfoodfacts.org/images/products/611/103/500/0430/


This is the base folder where product images like `front_en.104.400.jpg` are stored.

### Why
- Helps developers dynamically generate valid product image URLs.
- Avoids hardcoding full image filenames (which vary by language, view, and version).
- Improves SDK flexibility for image access.

### Fixes bug(s)
- #615

### Part of 
- Enhancement of media/image support for developers using the SDK.

ShaunakMishra25 committed:
chore: remove test code after verifying getProductImageFolder

VaiTon commented on src/main.ts:
```diff
@@ -61,6 +61,13 @@ export class OpenFoodFacts {
+
+    return `https://images.openfoodfacts.org/images/products/${chunks.join("/")}/`;
+  }
+
```
Move this method under the constructor, preferably at the end of the file

ShaunakMishra25 replied:
Thanks for the feedback, @VaiTon!

 I’ve extracted the image URL prefix to a `const IMAGE_BASE_URL` outside the class.  
 Moved the method to after the constructor as requested.

 ##As for the use case:
This method helps developers dynamically build the image folder path for any product barcode. It avoids hardcoding full image filenames (like `front_en.400.jpg`) and instead points to the correct folder where all image variations are stored. Useful for apps that display or fetch images on demand.

Let me know if there's anything else to refine — happy to adjust!

VaiTon reviewed (commented):
Hey, thank you for the contribution! Just two quick things:
- what is the usecase of this method?
- please extract the URL prefix to a const outside of the class

ShaunakMishra25 committed:
refactor: move getProductImageFolder and extract IMAGE_BASE_URL constant

ShaunakMishra25 commented:
Hey! I accidentally clicked “Resolve” earlier — sorry about that! I've now made the changes (moved the method after constructor and extracted the image URL prefix). Let me know if anything else needs tweaking

VaiTon reviewed (commented):
Please move the const to https://github.com/openfoodfacts/openfoodfacts-nodejs/blob/develop/src/consts.ts

Also, would you mind adding a testcase for this function?

ShaunakMishra25 committed:
feat: add getProductImageFolder with test and const

VaiTon committed:
Refactor getProductImageFolder and update export

- Move getProductImageFolder to a standalone exported function - Update
logic to split product code into 3-digit chunks, with the last 4 digits
as a single chunk - Rename test file to off.spec.ts for consistency

VaiTon committed:
Fix formatting in consts.ts

ShaunakMishra25 committed:
Merge branch 'develop' into feat/product-image-folder

VaiTon committed:
Fix test, use getProductImageFolder directly

ShaunakMishra25 commented:
Thanks for the review and merge! Glad I could contribute. Looking forward to helping out more in the future
